### PR TITLE
orogene 0.3.31 (new formula)

### PR DESCRIPTION
**Pull request by jasikpark**
initial attempt at contributing an orogen formula to Homebrew

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [X] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/o/orogene.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    sha256 cellar: :any_skip_relocation, arm64_sonoma:   "d76286dbec347fd26739d5e19742c54b30fc7616a780e499812db36a2c3b8102"
+    sha256 cellar: :any_skip_relocation, arm64_ventura:  "e30c0cf05dbcb16a4a5820126dd105067974e3a82d786a07fd0e5b2f36fb8085"
+    sha256 cellar: :any_skip_relocation, arm64_monterey: "e30c0cf05dbcb16a4a5820126dd105067974e3a82d786a07fd0e5b2f36fb8085"
+    sha256 cellar: :any_skip_relocation, arm64_big_sur:  "e30c0cf05dbcb16a4a5820126dd105067974e3a82d786a07fd0e5b2f36fb8085"
+    sha256 cellar: :any_skip_relocation, sonoma:         "f00c87a3ff4f645c600903f0ea7861f3629a10559b60ca9215e71135d427554c"
+    sha256 cellar: :any_skip_relocation, ventura:        "14919101e5f7d51d091463a83ab2adad69e808991fc6e957ff5c6b7e46f9d85c"
+    sha256 cellar: :any_skip_relocation, monterey:       "14919101e5f7d51d091463a83ab2adad69e808991fc6e957ff5c6b7e46f9d85c"
+    sha256 cellar: :any_skip_relocation, big_sur:        "14919101e5f7d51d091463a83ab2adad69e808991fc6e957ff5c6b7e46f9d85c"
+    sha256 cellar: :any_skip_relocation, x86_64_linux:   "d0c8001ffac28e44610535a60fd5a3bca5b649850f2360a2fec2676f7b2255a0"
+  end
```
Comment (p-linnane) [inline Formula/o/orogene.rb:20]:
```suggestion

```

File: Formula/o/orogene.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  depends_on "rust" => :build
+  depends_on "openssl"
```
Comment (p-linnane) [inline Formula/o/orogene.rb:23]:
```suggestion
  depends_on "openssl@3"
```

File: Formula/o/orogene.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
+class Orogene < Formula
+  desc "`node_modules/` package manager and utility toolkit"
+  homepage "https://orogene.dev"
+  url "https://github.com/orogene/orogene/archive/refs/tags/v0.3.31.tar.gz"
+  version "0.3.31"
```
Comment (p-linnane) [inline Formula/o/orogene.rb:5]:
```suggestion

```

**Review by p-linnane — CHANGES_REQUESTED** [review]

File: Formula/o/orogene.rb (hunk 2 above)
Comment (zkat) [reply Formula/o/orogene.rb:23]: I believe this should be @1

Comment (p-linnane) [reply Formula/o/orogene.rb:23]: OpenSSL 1.1.1 is end of life. We will not accept new formulae that require it.

**Comment by zkat** [comment]
Correction: I’m not sure orogene requires openssl to at all on Mac. Just on Linux. So you can probably remove that requirement altogether.

**Comment by zkat** [comment]
We already have a tap getting automatically published to https://github.com/orogene/homebrew-tap

does it make sense to use that? Is there automation around it?

**Comment by p-linnane** [comment]
> We already have a tap getting automatically published to https://github.com/orogene/homebrew-tap
> 
> does it make sense to use that? Is there automation around it?

The main `homebrew-core` repository is the default for Homebrew, which has a much wider reach than upstream taps. There's no harm in hosting your own, but it would not prevent us from packaging it in the main repo. One benefit is your users can run `brew install orogen` rather than specifying a tap.

A popular Action that many projects integrate into their release process is https://github.com/mislav/bump-homebrew-formula-action. It will open a PR with the version bump when you cut a release.

**Review by p-linnane — APPROVED** [review]

**Comment by jasikpark** [comment]
what build requirement do i need on linux?? https://github.com/Homebrew/homebrew-core/actions/runs/6364862360/job/17281469931#step:4:1

do i need to specify openssl for both build and runtime?

**Comment by p-linnane** [comment]
I've pushed a change to fix the build on Linux. This should be good to go now.

**Review by p-linnane — APPROVED** [review]

**Review by chenrui333 — APPROVED** [review]

**Comment by chenrui333** [comment]
Thanks folks, merging!

**Comment by zkat** [comment]
so this means I should get rid of the tap at orogene/homebrew-tap, correct? Are there docs for doing this redirect you mentioned?

**Comment by p-linnane** [comment]
@zkat I've opened a PR for the migration: https://github.com/orogene/homebrew-tap/pull/1

File: Formula/o/orogene.rb [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  on_linux do
+    depends_on "openssl@3"
+  end
+
+  def install
+    # Ensure that the `openssl` crate picks up the intended library.
+    ENV["OPENSSL_DIR"] = Formula["openssl@3"].opt_prefix
+    ENV["OPENSSL_NO_VENDOR"] = "1"
+
+    system "cargo", "install", *std_cargo_args
```
Comment (zkat) [inline Formula/o/orogene.rb:31]: @jasikpark sorry, I guess I didn't look closely enough: We actually automatically publish prebuilt binaries for apple x86, apple arm, and linux x86, so you don't need to do a full-on cargo install (and require build deps) if you just pull those down: https://github.com/orogene/orogene/releases/tag/v0.3.32

Is that acceptable for homebrew?

Comment (p-linnane) [reply Formula/o/orogene.rb:31]: Everything in `homebrew-core` must be built from source.

**Pull request merged**